Week1/homework/world-queries.txt

```diff
@@ -37,4 +37,190 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
+10 rows in set (0.01 sec)
+
+5- What's the top 10 most populated cities?
+
+SELECT name, population FROM city ORDER BY population DESC limit 10;
++-----------------------+------------+
+| name                  | population |
++-----------------------+------------+
+| Mumbai (Bombay)       |   10500000 |
+| Seoul                 |    9981619 |
+| Sï¿½o Paulo           |    9968485 |
+| Shanghai              |    9696300 |
+| Jakarta               |    9604900 |
+| Karachi               |    9269265 |
+| Istanbul              |    8787958 |
+| Ciudad de Mï¿½xico    |    8591309 |
+| Moscow                |    8389200 |
+| New York              |    8008278 |
++-----------------------+------------+
+10 rows in set (0.01 sec)
+
+6- Select the top 3 countries by population that their names start with the letter ‘P’
+
+SELECT name, population FROM country where name LIKE 'P%'  ORDER BY population DESC limit 3;
++-------------+------------+
+| name        | population |
++-------------+------------+
+| Pakistan    |  156483000 |
+| Philippines |   75967000 |
+| Poland      |   38653600 |
++-------------+------------+
+3 rows in set (0.01 sec)
+
+7- What is the population in Asia?
+
+SELECT continent, sum(population) FROM country WHERE continent = 'Asia';
++-----------+-----------------+
+| continent | sum(population) |
++-----------+-----------------+
+| Asia      |      3705025700 |
++-----------+-----------------+
+1 row in set (0.00 sec)
+
+8- Which languages are spoken in the Region of ‘South America’?
+
+SELECT language FROM countrylanguage JOIN country ON countrylanguage.countrycode = country.code where continent = 'South America';
++------------------+
+| language         |
++------------------+
+| Indian Languages |
+| Italian          |
+| Spanish          |
+| Aimarï¿½         |
+| Guaranï¿½        |
+| Ketï¿½ua         |
+| Spanish          |
+| German           |
+| Indian Languages |
+| Italian          |
+| Japanese         |
+| Portuguese       |
+| Aimarï¿½         |
+| Araucan          |
+| Rapa nui         |
+| Spanish          |
+| Arawakan         |
+| Caribbean        |
+| Chibcha          |
+| Creole English   |
+| Spanish          |
+| Ketï¿½ua         |
+| Spanish          |
+| English          |
+| Creole French    |
+| Indian Languages |
+| Arawakan         |
+| Caribbean        |
+| Creole English   |
+| Aimarï¿½         |
+| Ketï¿½ua         |
+| Spanish          |
+| German           |
+| Guaranï¿½        |
+| Portuguese       |
+| Spanish          |
+| Hindi            |
+| Sranantonga      |
+| Spanish          |
+| Goajiro          |
+| Spanish          |
+| Warrau           |
++------------------+
+42 rows in set (0.01 sec)
+
+10- What are the languages spoken on all cities named ‘Barcelona’ (you may need to join 3 tables =) )?
+
+SELECT language FROM countrylanguage JOIN country ON countrylanguage.countrycode = country.code JOIN city ON country.code = city.countrycode where city.name = 'Barcelona';
```

Comment:
You can use `SELECT distinct language...` in this query. That way "Spanish" wouldn't be repeated twice in the results.

Reply:
Yep, thanks